[DOC-779] Added metrics collection level (#28677)
yugabyte/yugabyte-db: docs/content/stable/yugabyte-platform/alerts-monitoring/anywhere-metrics.md

For more information, refer to [Querying Prometheus](https://prometheus.io/docs/prometheus/latest/querying/basics/) in the Prometheus documentation.

## Metrics collection level

> The metrics collection level setting controls how many database metrics are collected and stored. It helps balance system performance with monitoring needs, allowing administrators to optimize resource usage based on their specific observability requirements.

Reviewer: ```suggestion
The metrics collection level setting controls how many database metrics are collected and stored. It helps balance system performance with monitoring needs, so that you can optimize resource usage based on your specific observability requirements.
```

> The metrics collection level can be set via the **Metrics Collection Level** Universe Runtime Configuration option (config key `yb.metrics.collection_level`) to true. Refer to [Manage runtime configuration settings](../../../yugabyte-platform/administer-yugabyte-platform/manage-runtime-config/). You must be a Super Admin to set universe runtime configuration flags.

Reviewer 2: Remove 'to true'.
Also - you don't need to be a super admin to set universe level configuration.

> The metrics collection level can be set via the **Metrics Collection Level** Universe Runtime Configuration option (config key `yb.metrics.collection_level`) to true. Refer to [Manage runtime configuration settings](../../../yugabyte-platform/administer-yugabyte-platform/manage-runtime-config/). You must be a Super Admin to set universe runtime configuration flags.

Reviewer: ```suggestion
Set the metrics collection level using the **Metrics Collection Level** Universe Runtime Configuration option (config key `yb.metrics.collection_level`). Refer to [Manage runtime configuration settings](../../../yugabyte-platform/administer-yugabyte-platform/manage-runtime-config/).
```

> The flag supports four distinct collection levels as described in the following table:

Reviewer: ```suggestion
You can set the collection level to the following:
```

| Level | Description |
|:-------|:---------|

> | ALL | Collects all metrics from the database. |

Reviewer 2: Would make sense to add a notice for ALL level that it's not recommended to turn it on for large databases with a lot of tables - as it puts really high load on the YBA Prometheus and can cause OOMs on YBA node.

> | NORMAL (default) | Collects all server-level metrics, and essential table-level metrics (required for YugabyteDB Anywhere UI and alerts). |

Reviewer 2: I think we recently added another option called TABLE_OFF as part of https://yugabyte.atlassian.net/browse/PLAT-18210
And it's backported to 2025.1 and 2024.2 releases.
The option basically means we only scrape server level metrics.
Would be nice to add it right away if we're making a change, but only for above releases.

| MINIMAL | Collects limited list of server-level metrics (required for YugabyteDB Anywhere UI and alerts), and very limited table-level metrics (required for alerts and table size display). |

> | OFF | Completely disables metric collection. |

Reviewer: ```suggestion
| OFF | Disables metric collection. |
```
